frontend/server/handlers/pod-logs.ts

```diff
@@ -73,7 +73,16 @@ export function getPodLogsHandler(
 
     try {
       const stream = await getPodLogsStream(podName, podNamespace);
-      stream.on('error', err => res.status(500).send('Could not get main container logs: ' + err));
+      stream.on('error', err => {
+        if (
+          err?.message &&
+          err.message?.indexOf('Unable to find pod log archive information') > -1
+        ) {
+          res.status(404).send('pod not found');
```

Comment:
nit: also append `err` to make sure we can debug if there are false-positives of 404s?

Reply:
ohhh, I forgot I approved before.
Feel free to update it in a following up PR